threediresultstyler/threediresultstyler_dockwidget.py

```diff
@@ -0,0 +1,259 @@
+
+        for layer in schematisation_group.findLayers():
+            settings['layer'][layer.name()] = {}
+            settings['layer'][layer.name()]['visible'] = layer.isVisible()
+            settings['layer'][layer.name()]['expanded'] = layer.isExpanded()
+            settings['layer'][layer.name()]['subset'] = layer.layer().subsetString()
+
+        settings_dir = os.path.join(os.path.dirname(__file__), 'tree_settings')
+        filename = QtGui.QFileDialog.getSaveFileName(self, 'Open file', settings_dir,"JSON files (*.json)")[0]
+
+
+        with open(str(filename), 'w', encoding='utf-8') as outfile:
+            json.dump(settings, outfile, ensure_ascii=False, indent=2)
+            
+    def load_tree_settings(self):
+        settings_dir = os.path.join(os.path.dirname(__file__), 'tree_settings')
+        filename = QtGui.QFileDialog.getOpenFileName(self, 'Open file', settings_dir,"JSON files (*.json)")[0]
+        
+        root = QgsProject.instance().layerTreeRoot()
+        schematisation_group = root.findGroup('schematisation')
+        
+        with open(filename) as json_file:  
+            settings = json.load(json_file)
+
+        groups = schematisation_group.findGroups()
+        layers = schematisation_group.findLayers()
+        
+        for group in groups:
+            try:
+                group.setItemVisibilityChecked(settings['group'][group.name()]['visible'])
+                group.setExpanded(settings['group'][group.name()]['expanded'])
+            except:
+                self.iface.messageBar().pushMessage("Warning", "Couldn't find settings for group {}".format(group.name()), level=Qgis.Warning)
+            
+        for layer in layers:
+            try:
+                layer.setItemVisibilityChecked(settings['layer'][layer.name()]['visible'])
+                layer.setExpanded(settings['layer'][layer.name()]['expanded'])
+                layer.layer().setSubsetString(settings['layer'][layer.name()]['subset'])
+            except:
+                self.iface.messageBar().pushMessage("Warning", "Couldn't find settings for layer {}".format(layer.name()), level=Qgis.Warning)
+            
+    def add_sqlite_connection(self):
+        layer = QgsProject.instance().mapLayersByName("v2_global_settings")[0]
+        source =Path(layer.dataProvider().dataSourceUri().split(' ')[0].split('=')[1][1:-1])
+        settings = QSettings()
+        #remove old connection
+        for key in settings.allKeys():
+            if "SpatiaLite/connections/3Di plugin" in key:
+                settings.remove(key)
+
+        settings.setValue("SpatiaLite/connections/%s/sqlitepath" %("3Di plugin: {}, {}".format(source.name, datetime.now().strftime("%Y-%m-%d %H:%M"))), str(source))
+        self.iface.reloadConnections()
+        
+    def highlight_courant(self):
+        line_results = QgsProject.instance().mapLayersByName("line_results")[0]
+        if line_results is not None:
```

Comment:
Je kan ook zoiets doen:

    if line_results is None:
        messsagebar("aargh")
        return
    # De eigenlijke code kan dan weer lekker 4 spaties naar links.
    # De uitzondering is dan de uitzondering ("if"), de hoofdcode "loopt gewoon door".

Reply:
fixed